src/api.mjs

```diff
@@ -121,6 +126,20 @@ router.post('/check', async (ctx) => {
     return
   }
 
+  // Validate viewport preset
+  if (viewport && !Object.keys(VIEWPORT_PRESETS).includes(viewport)) {
+    ctx.status = 400
+    ctx.body = { error: 'Bad Request: unknown viewport preset' }
+    return
+  }
+
+  // Ensure device and viewport are not both set (mutually exclusive)
+  if (device && viewport) {
+    ctx.status = 400
+    ctx.body = { error: 'Bad Request: cannot specify both device and viewport' }
+    return
+  }
```

Comment:
would be nice to also, clientside, unset device when selecting a viewport option and vice versa